airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -545,17 +544,17 @@ def monitor_launched_pod(self, launcher, pod) -> Tuple[State, Optional[str]]:
         :return:
         """
         try:
-            (final_state, result) = launcher.monitor_pod(pod, get_logs=self.get_logs)
+            (final_state, pod_info, result) = launcher.monitor_pod(pod, get_logs=self.get_logs)
         finally:
             if self.is_delete_operator_pod:
                 launcher.delete_pod(pod)
         if final_state != State.SUCCESS:
             if self.log_events_on_failure:
-                for event in launcher.read_pod_events(pod).items:
+                for event in pod_info.items:
```

Comment:
```suggestion
                for event in launcher.read_pod_events(pod).items:
```

Reply:
The problem here is that we've already deleted the pod on like 550, so we can't call `read_pod_events` against it

Reply:
The pod events are still available after deleting the pod. We can't use `read_namespaced_pod`, but `read_pod_events` should still work. We also won't get `pod_info` back here in all cases.

Reply:
Ah, that makes sense then.  I was just going over the code and your other comments and I wonder if it wouldn't be easier and better to re-read the pod like `self.pod = launcher.read_pod(self.pod)` in the `finally` blocks of `monitor_launched_pod` and `create_new_pod_for_operator`:
https://github.com/apache/airflow/blob/main/airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py#L528
https://github.com/apache/airflow/blob/main/airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py#L550

and then just log `self.pod` here:
https://github.com/apache/airflow/blob/main/airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py#L369

Way fewer changes and a lot less passing the pod around through the various functions.

Reply:
That works for me, just make sure any exceptions when reading it again don't bubble up.

Reply:
I was poking at this last night and I didn't end up being overly keen on overwriting the original pod spec in `self.pod` with the remote pod after completion.  I went back to the original idea in the PR, but I renamed all the `pod_info` to `remote_pod` - going with the `pod` variable name as you suggested ended up conflating the returned pod with the `pod` parameter passed into some functions and `self.pod` which just seemed confusing.

If you wouldn't mind taking another look at the PR it would be much appreciated :)

Reply:
Yeah, I went back and forth on that yesterday as well. There is precedence for setting `self.pod` to be the remote pod, but I think its easier to grok as it is now.

Thanks for being receptive to iterating on this 🍺

Reply:
> Thanks for being receptive to iterating on this
No problem I think we all want a solution that everyone likes 😄